### PR TITLE
feat(material-experimental/mdc-form-field): Add option for dynamic su…

### PR DESCRIPTION
**Pull request by kseamon**
…bscript height based on number of hints and errors.

### CODE REVIEW

File: src/material-experimental/mdc-form-field/_form-field-subscript.scss [hunk 1]
```diff
@@ -16,14 +16,20 @@
...
   .mat-mdc-form-field-error-wrapper {
-    position: absolute;
-    top: 0;
-    left: 0;
-    right: 0;
     padding: 0 mdc-textfield-variables.$padding-horizontal;
   }
 
-  .mat-mdc-form-field-bottom-align::before {
+  .mat-mdc-form-field-bottom-align:not(.mat-mdc-form-field-subscript-dynamic-size) {
```
Comment (mmalerba) [inline src/material-experimental/mdc-form-field/_form-field-subscript.scss:22]: We'll see how the presubmit goes, but I worry about the specificity increase. An alternate approach would be to add a stronger rule for the dynamic class and set `position: static`

**Review by mmalerba — APPROVED** [review]

File: src/material-experimental/mdc-form-field/_form-field-subscript.scss (hunk 1 above)
Comment (kseamon) [reply src/material-experimental/mdc-form-field/_form-field-subscript.scss:22]: If the presubmit fails due to this, I'll change it. I took a quick look and did not see anything that would conflict with it.

**Review by devversion — APPROVED** [review]
LGTM, just some minor comments.

I also wanted to add that I think that the dynamic mode could become a cause for layout shifts if e.g. hints/errors are appearing, so I assume that is something, when explicitly enabled, that is left to the user (but might be worth mentioning somewhere?)

**Comment by kseamon** [comment]
> 

I added a note about layout shift to the JSDoc. I'll make sure it goes into the internal docs that will later get copied out to public docs when MDC versions take over.

File: src/material-experimental/mdc-form-field/_form-field-subscript.scss (hunk 1 above)
Comment (kseamon) [reply src/material-experimental/mdc-form-field/_form-field-subscript.scss]: I did end up having to fix the selectors - passing now

**Pull request merged**